### ISSUE TITLE
Notification components stylesheet class attribute should be copied to container element.

### ISSUE THREAD

**Issue by stefanuebe** [vaadin/flow#3844]
I extended the Vaadin Notification component (Java) and added a single css class plus some sub components.

```
@StyleSheet("... styles.css")
public class NotificationExtension extends Notification {
   public NotificationExtension() {
      addClassName("some-name");

      Div caption = new Div();
      caption.addClassName("caption");

      add(caption);
   }
}
```

The css code is simple as
```
.some-name .caption {
   color: red;
}
```

As a result in html there is an element vaadin-notification with the given class name "some-name" and a siblin vaadin-notification-container without the class name. The container contains the caption element. 

Since the vn container does not have the class name, the css styles are not recognized. 

Request: Necessary attributes like class name should be copied to the vn container element, so that stylings will be recognized in a use case like this (without the usage of polymer).

**Comment by jouni** [comment]
This should be implemented in https://github.com/vaadin/vaadin-notification and all components that have use vaadin-overlay internally (dialog, date picker, combo box, dropdown menu, context menu).

**Comment by pleku** [comment]
To my understanding, this has been fixed for all `v-overlay` based web components, should just be enough to verify this, preferably add a test and close the issue.

**Comment by web-padawan** [comment]
See https://github.com/vaadin/vaadin-notification/releases/tag/v1.0.0-beta4
Actually, notification no longer depends on the overlay, but the approach is the same.

**Comment by ZheSun88** [comment]
In this case, this issue is not blocked by another ticket any more, right?

**Comment by pleku** [comment]
>while no classname for the container

There is no class name, but it doesn't need one since the styles are copied.

**Comment by ZheSun88** [comment]
More tests on this issue.

- Test One
I have added a component inside the notification and .css file which contains the same setting as in the original isssue `color: red`, what i get is still a un-colored one, 
![image](https://user-images.githubusercontent.com/31067185/39170990-25511f1a-47a6-11e8-819a-cb1cbaacf124.png)

- Test Two
I have added some text in side the notification (interHTML), and .css file looks like 
```
.some-name {
	color: blue;
}
```
what i get is 
![image](https://user-images.githubusercontent.com/31067185/39171242-cea2c4ce-47a6-11e8-9c60-57109a75f99e.png)

I can see that all the styles are in `vaadin-notification` but it isnt copied to the `vaadin-notification-container`
![image](https://user-images.githubusercontent.com/31067185/39171479-9f511274-47a7-11e8-8778-a0298878b313.png)

@web-padawan @jouni  is there something that i used in a wrong way, or the issue is still existing?
i am using the newest version of `vaadin-notification`

**Comment by web-padawan** [comment]
Ok, apparently there is a misunderstandig. The styles are applied to the content of the template only.
So, for the "Custom Styling of the Content" demo [here](https://cdn.vaadin.com/vaadin-notification/1.0.0-beta4/demo/#notification-basic-demos), it looks like this:

![screen shot 2018-04-24 at 10 36 26](https://user-images.githubusercontent.com/10589913/39172796-7f698a8c-47ab-11e8-9c9c-adbc78fa5134.png)

Prior to v1.0.0-beta4, it wasn't possible to apply the styles like this. Note the additional shadow root inserted into a `<div part="content">` - this is where the content and the corresponding `<style>` tag is getting inserted under native Shadow DOM implementation.

The class name from the `<vaadin-notification>` component itself is not copied, same applies to the `<vaadin-dialog>`. Was this supposed to be implemented as well?

Theoretically, it should be doable by propagating these classes to the `<div part="content">` mentioned above, but that could cause a mess in Shady mode.

/cc @tomivirkki

**Comment by ZheSun88** [comment]
okay, if i understand correctly, after beta4, `flow-component-render` is not inside the `overlay` part any more, so in this case, we cannot do the trick to style it as in the demo? 
as it is shown in the pic, 
![image](https://user-images.githubusercontent.com/31067185/39173502-8ec8fdbc-47ad-11e8-8b70-f996deaba352.png)
no matter the plain text, or a component, it is outside of the shadow root of vn container

**Comment by web-padawan** [comment]
In the overlay, we now have the `content` property referencing to the  content container after the template is stamped, see it [here](https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-overlay.html#L191). You should be able to attach the component renderer inside it, if this is how it is supposed to work.

This is what we are likely missing from the notification (reference to the content container not exposed as a public API). It is currently "protected" and stored as a `_cardContent`, see [here](https://github.com/vaadin/vaadin-notification/blob/master/src/vaadin-notification.html#L304).

Can you check if using that one for the component rendered could solve the problem, or not?

**Comment by gilberto-torrezan** [comment]
@web-padawan We don't have access to the overlay, since it is created outside the control of Flow. Is there a way to select the overlay (or the card) in the DOM created by a notification? Something like a `notification.card` or `notification.getOpenedCard` ? In that way we could find the right element in the DOM and then copy whatever class names are needed to the container.

**Comment by web-padawan** [comment]
@gilberto-torrezan as I mentioned above, the DOM node for the card content wrapper is stored in the protected property `_cardContent`. But the classes should not be applied to that wrapper, but for some node in its Shadow Root. So in order to copy the styles as you want, we would have to do the following steps:

1. create yet another `div`
2. stamp the notification template into it
3. append this div to shadow root of `_cardContent`
4. expose the reference to that `div`

This, from my point of view, introduces more complexity, and is unlikely to work in all cases, e. g. consider user attempting to apply the following CSS:
```css
vaadin-notification.some-name .caption {
   color: red;
}
```
The tag name selector won't apply for obvious reasons, while the simple class selector would (because of this "magical" copying). That would produce some kind of inconsistence.

**Comment by pleku** [comment]
Will be handled by https://github.com/vaadin/vaadin-notification-flow/issues/43 and https://github.com/vaadin/vaadin-notification-flow/issues/44.
